kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-data-sources/src/main/java/org/jbpm/springboot/datasources/JBPMDataSourceAutoConfiguration.java

```diff
@@ -138,11 +142,16 @@ private XADataSource createXaDataSourceInstance(String className) {
     }
 
     private void bindXaProperties(XADataSource target, DataSourceProperties properties) {
-        MutablePropertyValues values = new MutablePropertyValues();
-        values.add("user", properties.determineUsername());
-        values.add("password", properties.determinePassword());
-        values.add("url", properties.determineUrl());
-        values.addPropertyValues(properties.getXa().getProperties());
-        new RelaxedDataBinder(target).withAlias("user", "username").bind(values);
+        MapConfigurationPropertySource values = new MapConfigurationPropertySource();
+        values.put("user", properties.determineUsername());
+        values.put("password", properties.determinePassword());
+        values.put("url", properties.determineUrl());
+        values.putAll(properties.getXa().getProperties());
+        
+        ConfigurationPropertyNameAliases aliases = new ConfigurationPropertyNameAliases();
+        aliases.addAliases("user", "username");
+        
+        Binder binder = new Binder(values.withAliases(aliases));
+        binder.bind(ConfigurationPropertyName.EMPTY, Bindable.ofInstance(target));
```

Comment:
This EMPTY means it will apply all properties possible?

Reply:
that's how I understand it as well.